source/gameengine/GameLogic/SCA_VibrationActuator.cpp

```diff
@@ -0,0 +1,143 @@
+/*
+* ***** BEGIN GPL LICENSE BLOCK *****
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
+* as published by the Free Software Foundation; either version 2
+* of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*
+* The Original Code is Copyright (C) 2001-2002 by NaN Holding BV.
+* All rights reserved.
+*
+* The Original Code is: all of this file.
+*
+* Contributor(s): none yet.
+*
+* ***** END GPL LICENSE BLOCK *****
+*/
+
+/** \file gameengine/Ketsji/SCA_VibrationActuator.cpp
+*  \ingroup ketsji
+*/
+
+
+#include "SCA_VibrationActuator.h"
+#include "SCA_JoystickManager.h"
+#include "DEV_JoystickPrivate.h"
+#include <iostream> //std::cout
+#ifdef WITH_SDL
+	#include "SDL.h"
+#endif // WITH_SDL
+
+SCA_VibrationActuator::SCA_VibrationActuator(SCA_IObject* gameobj, int joyindex, float strength, int duration)
+	: SCA_IActuator(gameobj, KX_ACT_VIBRATION),
+	m_joyindex(joyindex),
+	m_strength(strength),
+	m_duration(duration)
+{
+}
+
+SCA_VibrationActuator::~SCA_VibrationActuator(void)
+{
+}
+
+CValue* SCA_VibrationActuator::GetReplica(void)
+{
+	SCA_VibrationActuator* replica = new SCA_VibrationActuator(*this);
+	replica->ProcessReplica();
+	return replica;
+}
+
+bool SCA_VibrationActuator::Update()
+{
+#ifdef WITH_SDL
+
+	/* Can't init instance in constructor because m_joystick list is not available yet */
+	SCA_JoystickManager *mgr = (SCA_JoystickManager *)GetLogicManager();
+	DEV_Joystick *instance = mgr->GetJoystickDevice(m_joyindex) ? mgr->GetJoystickDevice(m_joyindex) : NULL;
+
+	if (!instance) {
+		return false;
+	}
+	bool bNegativeEvent = IsNegativeEvent();
+
+	if (bNegativeEvent) {
+		return false;
+	}
+	
+	SDL_Haptic *haptic;
```

Comment:
To rejoin lordloki I think you can create a `DEV_Joystick::RumblePlay(strength, duration)` to avoid include SDL here.